Sources/Subprocess/IO/Output.swift

```diff
@@ -151,50 +137,34 @@ public struct StringOutput<Encoding: Unicode.Encoding>: OutputProtocol {
 /// A concrete `Output` type for subprocesses that collects output
 /// from the subprocess as `[UInt8]`. This option must be used with
 /// the `run()` method that returns a `CollectedResult`
-#if SubprocessSpan
-@available(SubprocessSpan, *)
-#endif
 public struct BytesOutput: OutputProtocol {
     public typealias OutputType = [UInt8]
     public let maxSize: Int
 
     internal func captureOutput(
         from diskIO: consuming TrackedPlatformDiskIO?
     ) async throws -> [UInt8] {
-        var diskIOBox: TrackedPlatformDiskIO? = consume diskIO
-        return try await withCheckedThrowingContinuation { continuation in
-            let _diskIO = diskIOBox.take()
-            guard let _diskIO = _diskIO else {
-                // Show not happen due to type system constraints
-                fatalError("Trying to capture output without file descriptor")
-            }
-            _diskIO.readUntilEOF(upToLength: self.maxSize) { result in
-                switch result {
-                case .success(let data):
-                    // FIXME: remove workaround for
-                    // rdar://143992296
-                    // https://github.com/swiftlang/swift-subprocess/issues/3
-                    #if os(Windows)
-                    continuation.resume(returning: data)
-                    #else
-                    continuation.resume(returning: data.array())
-                    #endif
-                case .failure(let error):
-                    continuation.resume(throwing: error)
-                }
-            }
+        guard let diskIO = diskIO else {
+            // Show not happen due to type system constraints
```

Comment:
If this shouldn't happen due to type system constraints, is there a way to make that expressed in the type system instead of a fatal error at runtime?